DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -96,18 +97,23 @@ public IActionResult Index(DlsSubApplication dlsSubApplication)
             var centreId = User.GetCentreId();
             var userId = User.GetUserIdKnownNotNull();
             var userEntity = userService.GetUserById(userId)!;
-            var delegateAccount =
-                userEntity.DelegateAccounts.SingleOrDefault(da => da.CentreId == centreId);
+            var delegateAccount = userEntity.GetCentreAccountSet(centreId)?.DelegateAccount;

```

Comment:
I'm getting a warning about a possible NullReferenceException on the userEntity used in this line, despite the forgiving operator being applied above. Same in the Index. Weird.

Reply:
Yeah that looks to have solved it.